src/Microsoft.IdentityModel.JsonWebTokens/Json/JsonClaimSet.cs

```diff
@@ -231,32 +181,27 @@ internal T GetValue<T>(string key)
         /// </summary>
         /// <typeparam name="T"></typeparam>
         /// <param name="key"></param>
-        /// <param name="throwEx"></param>
+        /// <param name="throwEx">if this is called from TryGetValue then we don't want to throw.</param>
         /// <param name="found"></param>
         /// <returns></returns>
         internal T GetValue<T>(string key, bool throwEx, out bool found)
         {
             found = _jsonClaims.TryGetValue(key, out object obj);
 
             if (!found)
-            {
-                if (throwEx)
-                    throw LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.IDX14304, key)));
-                else
-                    return default;
-            }
+                return throwEx ? throw LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.IDX14304, key))) : default;
 
             if (obj == null)
                 if (typeof(T) == typeof(object) || typeof(T).IsClass || Nullable.GetUnderlyingType(typeof(T)) != null)
-                {
                     return (T)(object)null;
-                }
                 else
-                {
-                    found = false;
                     return default;
-                }
 
+            // When the JsonClaimSet is created JsonArray and JsonObject are stored as JsonElement's
+            if (obj is JsonElement jsonElement)
```

Comment:
suggestion:

If we are okay leaking JsonElement as a type, we could write this as

```cs
            if (obj is JsonElement jsonElement)
            {
                if (typeof(T) == typeof(JsonElement))
                    return (T)(object)jsonElement;

                return (T)JsonSerializerPrimitives.CreateTypeFromJsonElement<T>(jsonElement);
            }
```

Which would be more flexible if someone wants to read/write arbitrarily complex JSON and we don't have to explicitly handle the type.

Thoughts @jennyf19 @brentschmaltz ?

Reply:
@keegan-caruso I moved the check if (typeof(T) == objType) => return (T)(obj), which is essentially what you are saying.